File: 2025/July-25-2025/twoSum.js
var nums = [3,4,5,6], target = 7

const twoSum = (nums, target) => {
  var hash = {}
  for(let i = 0; i < nums.length; i++){
      let remainder = target - nums[i]

      if(remainder in hash){
          return [hash[remainder], i]
      }
      hash[nums[i]] = i
  }
}


console.log(twoSum(nums, target))

File: 2025/October-31-2025/twoSum.py
from typing import List

nums = [3,4,5,6]
target = 7


def twoSum(nums: List[int], target: int) -> bool: 
  hash = {}

  for i in range(0, len(nums)): 
    remainder = target - nums[i]

    if remainder in hash: 
      return [hash[remainder], i]
    else: 
      hash[nums[i]] = i
  
print (twoSum(nums, target))